engine/src/main/java/org/hibernate/validator/internal/metadata/provider/AnnotationMetaDataProvider.java

```diff
@@ -220,7 +225,8 @@ public AnnotationProcessingOptions getAnnotationProcessingOptions() {
 				continue;
 			}
 
-			JavaBeanField javaBeanField = new JavaBeanField( field );
+			String name = new ExecutablePropertyNodeNameProvider( propertyNodeNameProvider, beanClass ).getName( field.getName() );
```

Comment:
We don't really need to create this `ExecutablePropertyNodeNameProvider`. We should have a wrapper of field/getter which implements `Property` interface that was added in this PR and pass it to the propertyNodeNameProvider. We even shouldn't add `propertyNodeNameProvider` as an additional field to this provider. We already have it in `JavaBeanHelper` so we probably should use it. We can add a method in that helper that would resolve a name for us.